BUG: stats: Two new test failures

With SCIPY_XSLOW set, I get two new failures in the `TestFit` class in `scipy/stats/tests/test_fit.py`.  Details below.

<details>
<summary>Terminal transcript</summary>

```
$ SCIPY_XSLOW=1 python3 dev.py test -t scipy.stats.tests.test_fit -m full -j 10
💻  ninja -C /home/warren/repos/git/forks/scipy/build -j12
ninja: Entering directory `/home/warren/repos/git/forks/scipy/build'
[2/2] Generating scipy/generate-version with a custom command
Build OK
💻  meson install -C build --only-changed
Installing, see meson-install.log...
Installation OK
SciPy from development installed path at: /home/warren/repos/git/forks/scipy/build-install/lib/python3.12/site-packages
Running tests for scipy version:1.14.0.dev0+1281.cd05b0e, installed at:/home/warren/repos/git/forks/scipy/build-install/lib/python3.12/site-packages/scipy
============================= test session starts ==============================
platform linux -- Python 3.12.3, pytest-8.1.1, pluggy-1.5.0
rootdir: /home/warren/repos/git/forks/scipy
configfile: pytest.ini
plugins: timeout-2.3.1, xdist-3.5.0, hypothesis-6.100.1, cov-5.0.0
10 workers [629 items]    
.xxxxxxxx.xx...x..xxx.........x.xxxxxxx..xxx.x...x.....x....x.xxx.xxxxxx [ 11%]
xx.x.....xx.........xx.xx..x.x..x..x....x.xxxx...xxxx.xx.xx.x..x..x..... [ 22%]
..........................x.........xxx.xx.............................. [ 34%]
..x.......x...x.x.x...............x.x..x...........x...xx..........x.... [ 45%]
.............x.x..........x.......x..................................... [ 57%]
..s.........s.....ss....s......................s.s...................... [ 68%]
..............ss...........F.....s...........ss..........s........sss... [ 80%]
......s................s..............s......s...s...................... [ 91%]
..............s.............F........................                    [100%]
=================================== FAILURES ===================================
______________________ TestFit.test_basic_fit_mse[argus] _______________________
[gw4] linux -- Python 3.12.3 /home/warren/py3.12.3/bin/python3
scipy/stats/tests/test_fit.py:533: in test_basic_fit_mse
    self.basic_fit_test(dist_name, "mse")
        dist_name  = 'argus'
        self       = <scipy.stats.tests.test_fit.TestFit object at 0x79b4b8dc41d0>
scipy/stats/tests/test_fit.py:524: in basic_fit_test
    assert_nlff_less_or_close(dist, data, res.params, ref, **self.tols,
        N          = 5000
        bounds     = array([[1.e-01, 1.e+01],
       [0.e+00, 1.e+01],
       [1.e-16, 1.e+01]])
        data       = array([4.94646235, 5.60692977, 5.69664841, ..., 5.41930797, 4.91919232,
       5.33154651])
        dist       = <scipy.stats._continuous_distns.argus_gen object at 0x79b4da500980>
        dist_data  = {'alpha': (3.570477051665046,), 'anglit': (), 'arcsine': (), 'argus': (1.0,), ...}
        dist_name  = 'argus'
        loc        = 4.790547883456026
        method     = 'mse'
        nlff_name  = '_penalized_nlpsf'
        nlff_names = {'mle': 'nnlf', 'mse': '_penalized_nlpsf'}
        ref        = [1.0, 4.790547883456026, 0.9992905124308905]
        res        =   params: FitParams(chi=4.9473425236047355, loc=0.004295788296504643, scale=5.784914586925937)
 success: True
 message: 'Optimization terminated successfully.'
        rng        = Generator(PCG64) at 0x79B4B8D48E40
        scale      = 0.9992905124308905
        self       = <scipy.stats.tests.test_fit.TestFit object at 0x79b4b8dc41d0>
        shapes     = array([1.])
        sup        = <numpy.testing._private.utils.suppress_warnings object at 0x79b4b67c0bf0>
scipy/stats/tests/test_fit.py:360: in assert_nlff_less_or_close
    np.testing.assert_allclose(nlff1, nlff0, rtol=rtol, atol=atol)
        atol       = 0.05
        data       = array([4.94646235, 5.60692977, 5.69664841, ..., 5.41930797, 4.91919232,
       5.33154651])
        dist       = <scipy.stats._continuous_distns.argus_gen object at 0x79b4da500980>
        nlff       = <bound method rv_generic._penalized_nlpsf of <scipy.stats._continuous_distns.argus_gen object at 0x79b4da500980>>
        nlff0      = 45517.83081787304
        nlff1      = 46035.24003829856
        nlff_name  = '_penalized_nlpsf'
        params0    = [1.0, 4.790547883456026, 0.9992905124308905]
        params1    = FitParams(chi=4.9473425236047355, loc=0.004295788296504643, scale=5.784914586925937)
        rtol       = 0.01
/home/warren/py3.12.3/lib/python3.12/contextlib.py:81: in inner
    return func(*args, **kwds)
E   AssertionError: 
E   Not equal to tolerance rtol=0.01, atol=0.05
E   
E   Mismatched elements: 1 / 1 (100%)
E   Max absolute difference: 517.40922043
E   Max relative difference: 0.01136718
E    x: array(46035.240038)
E    y: array(45517.830818)
        args       = (<function assert_allclose.<locals>.compare at 0x79b4b67ad120>, array(46035.2400383), array(45517.83081787))
        func       = <function assert_array_compare at 0x79b4cfd82fc0>
        kwds       = {'equal_nan': True, 'err_msg': '', 'header': 'Not equal to tolerance rtol=0.01, atol=0.05', 'verbose': True}
        self       = <contextlib._GeneratorContextManager object at 0x79b4cfda0620>
______________________________ TestFit.test_guess ______________________________
[gw4] linux -- Python 3.12.3 /home/warren/py3.12.3/bin/python3
scipy/stats/tests/test_fit.py:711: in test_guess
    assert_allclose(res.params, params, **self.tols)
        N          = 2000
        bounds     = [(2, 200), (0.7, 70), (1.2, 120), (0, 10)]
        data       = array([6, 7, 6, ..., 7, 7, 7])
        dist       = <scipy.stats._discrete_distns.nhypergeom_gen object at 0x79b4da52d6a0>
        params     = (20, 7, 12, 0)
        res        =   params: FitParams(M=17.0, n=6.0, r=10.0, loc=1.0)
 success: True
 message: 'Optimization terminated successfully.'
        rng        = Generator(PCG64) at 0x79B4B911D460
        self       = <scipy.stats.tests.test_fit.TestFit object at 0x79b4b671c5c0>
/home/warren/py3.12.3/lib/python3.12/contextlib.py:81: in inner
    return func(*args, **kwds)
E   AssertionError: 
E   Not equal to tolerance rtol=0.01, atol=0.05
E   
E   Mismatched elements: 4 / 4 (100%)
E   Max absolute difference: 3.
E   Max relative difference: 0.16666667
E    x: array([17.,  6., 10.,  1.])
E    y: array([20,  7, 12,  0])
        args       = (<function assert_allclose.<locals>.compare at 0x79b4b63f8900>, array([17.,  6., 10.,  1.]), array([20,  7, 12,  0]))
        func       = <function assert_array_compare at 0x79b4cfd82fc0>
        kwds       = {'equal_nan': True, 'err_msg': '', 'header': 'Not equal to tolerance rtol=0.01, atol=0.05', 'verbose': True}
        self       = <contextlib._GeneratorContextManager object at 0x79b4cfda0620>
=========================== short test summary info ============================
FAILED scipy/stats/tests/test_fit.py::TestFit::test_basic_fit_mse[argus] - AssertionError: 
FAILED scipy/stats/tests/test_fit.py::TestFit::test_guess - AssertionError: 
============ 2 failed, 516 passed, 22 skipped, 89 xfailed in 48.27s ============

```

</details>

<details>
<summary>Platform and version info</summary>

```
In [1]: import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info); scipy.show_config()
1.14.0.dev0+1281.cd05b0e 1.26.4 sys.version_info(major=3, minor=12, micro=3, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /usr/include/x86_64-linux-gnu/openblas-pthread/
    lib directory: /usr/lib/x86_64-linux-gnu/openblas-pthread/
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER=1 NO_CBLAS=
      NO_LAPACK= NO_LAPACKE=1 NO_AFFINITY=1 USE_OPENMP=0 generic MAX_THREADS=64
    pc file directory: /usr/lib/x86_64-linux-gnu/pkgconfig
    version: 0.3.20
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /usr/include/x86_64-linux-gnu/openblas-pthread/
    lib directory: /usr/lib/x86_64-linux-gnu/openblas-pthread/
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER=1 NO_CBLAS=
      NO_LAPACK= NO_LAPACKE=1 NO_AFFINITY=1 USE_OPENMP=0 generic MAX_THREADS=64
    pc file directory: /usr/lib/x86_64-linux-gnu/pkgconfig
    version: 0.3.20
  pybind11:
    detection method: config-tool
    include directory: unknown
    name: pybind11
    version: 2.12.0
Compilers:
  c:
    commands: cc
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  c++:
    commands: c++
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.0.10
  fortran:
    commands: gfortran
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  pythran:
    include directory: ../../../../../py3.12.3/lib/python3.12/site-packages/pythran
    version: 0.15.0
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
Python Information:
  path: /home/warren/py3.12.3/bin/python3.12
  version: '3.12'

```
</details>

--- a/scipy/stats/tests/test_fit.py
+++ b/scipy/stats/tests/test_fit.py
@@ -229,8 +229,8 @@ def test_nnlf_and_related_methods(dist, params):
 
 def cases_test_fit_mle():
     # These fail default test or hang
-    skip_basic_fit = {'argus', 'irwinhall', 'foldnorm', 'truncpareto', 
-                      'truncweibull_min', 'ksone', 'levy_stable', 
+    skip_basic_fit = {'argus', 'irwinhall', 'foldnorm', 'truncpareto',
+                      'truncweibull_min', 'ksone', 'levy_stable',
                       'studentized_range', 'kstwo', 'arcsine'}
 
     # Please keep this list in alphabetical order...
@@ -280,7 +280,8 @@ def cases_test_fit_mse():
                       'geninvgauss',  # quite slow (~4 minutes) but passes
                       'gausshyper', 'genhyperbolic',  # integration warnings
                       'tukeylambda',  # close, but doesn't meet tolerance
-                      'vonmises'}  # can have negative CDF; doesn't play nice
+                      'vonmises',  # can have negative CDF; doesn't play nice
+                      'argus'}  # doesn't meet tolerance; tested separately
 
     # Please keep this list in alphabetical order...
     slow_basic_fit = {'alpha', 'anglit', 'arcsine', 'betabinom', 'bradford',
@@ -545,7 +546,8 @@ class TestFit:
         res = stats.fit(dist, data, shape_bounds, optimizer=self.opt)
         assert_nlff_less_or_close(dist, data, res.params, shapes, **self.tols)
 
-    def test_argus(self):
+    @pytest.mark.parametrize("method", ('mle', 'mse'))
+    def test_argus(self, method):
         # Can't guarantee that all distributions will fit all data with
         # arbitrary bounds. This distribution just happens to fail above.
         # Try something slightly different.
@@ -555,7 +557,7 @@ class TestFit:
         shapes = (1., 2., 3.)
         data = dist.rvs(*shapes, size=N, random_state=rng)
         shape_bounds = {'chi': (0.1, 10), 'loc': (0.1, 10), 'scale': (0.1, 10)}
-        res = stats.fit(dist, data, shape_bounds, optimizer=self.opt)
+        res = stats.fit(dist, data, shape_bounds, optimizer=self.opt, method=method)
 
         assert_nlff_less_or_close(dist, data, res.params, shapes, **self.tols)
 
@@ -697,7 +699,7 @@ class TestFit:
         # Test that guess helps DE find the desired solution
         N = 2000
         # With some seeds, `fit` doesn't need a guess
-        rng = np.random.default_rng(1963904448561)
+        rng = np.random.default_rng(196390444561)
         dist = stats.nhypergeom
         params = (20, 7, 12, 0)
         bounds = [(2, 200), (0.7, 70), (1.2, 120), (0, 10)]

